### PR TITLE
Webview can accept specific file type

### PR DESCRIPTION
**Pull request by munir-rakuten**
# Description
Webview can accept specific image file type.

## Links
MINI-3346

# Checklist
- [x] I have read the [contributing guidelines](../.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](../.github/CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

**Review by khairul-alam-licon — APPROVED** [review]
Looks good 👍

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFileChooser.kt [hunk 1]
```diff
@@ -49,6 +49,26 @@ class MiniAppFileChooserDefault(var requestCode: Int) : MiniAppFileChooser {
             if (fileChooserParams?.mode == WebChromeClient.FileChooserParams.MODE_OPEN_MULTIPLE) {
                 intent?.putExtra(Intent.EXTRA_ALLOW_MULTIPLE, true)
             }
+            // Uses Intent.EXTRA_MIME_TYPES to pass multiple mime types.
+            fileChooserParams?.acceptTypes?.let { acceptTypes ->
+                if (!acceptTypes.equals("") && acceptTypes.size > 1) {
+                    // Accept all first.
+                    intent?.type = "*/*"
+                    val imageExtensionList = listOf(".jpg", ".jpeg", ".png")
```
Comment (corycaywood) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFileChooser.kt:57]: This is only going to work for `image/` mime types - also this still doesn't cover all image types such as svg, gif, etc. It's possible someone will want to use this for audio, video, or any other type of files as well. So we would need to update this function every time that we need to support another mime-type.

Could you instead use [MimeTypeMap.getMimeTypeFromExtension](https://developer.android.com/reference/android/webkit/MimeTypeMap#getMimeTypeFromExtension(java.lang.String)) to convert the extension to a mime type?

Comment (munir-rakuten) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFileChooser.kt]: @corycaywood updated the code for any type of files now..

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFileChooser.kt [hunk 2]
```diff
@@ -49,6 +50,17 @@ class MiniAppFileChooserDefault(var requestCode: Int) : MiniAppFileChooser {
             if (fileChooserParams?.mode == WebChromeClient.FileChooserParams.MODE_OPEN_MULTIPLE) {
                 intent?.putExtra(Intent.EXTRA_ALLOW_MULTIPLE, true)
             }
+            // Uses Intent.EXTRA_MIME_TYPES to pass multiple mime types.
+            fileChooserParams?.acceptTypes?.let { acceptTypes ->
+                if (!acceptTypes.equals("") && acceptTypes.size > 1) {
```
Comment (corycaywood) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFileChooser.kt:55]: `acceptTypes.size > 1` means that it won't work if only one type is passed, i.e. `accept=".jpg"`. Should it be like this instead?
```suggestion
                if (!acceptTypes.equals("") && acceptTypes.isNotEmpty()) {
```

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFileChooser.kt [hunk 3]
```diff
@@ -58,6 +70,32 @@ class MiniAppFileChooserDefault(var requestCode: Int) : MiniAppFileChooser {
...
+     * Validation Utility for MimeTypes.
+     * @param mimeTypes The Mimetypes needs to convert to valid types.
+     */
+    @VisibleForTesting
+    internal fun extractValidMimeTypes(mimeTypes: Array<String>): List<String> {
+        val results: MutableList<String> = ArrayList()
+
+        val mtm = MimeTypeMap.getSingleton()
+        for (mime in mimeTypes) {
+            if (mime.trim { it <= ' ' }.startsWith(".")) {
```
Comment (corycaywood) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFileChooser.kt:83]: Isn't a single space the default character for `trim`? So couldn't this be just `mime.trim()`?

Also, you're calling `mime.trim` several times so I think you should either set this to a variable or use something like `mime.trim().let { ... }`

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFileChooser.kt [hunk 4]
```diff
@@ -58,6 +70,32 @@ class MiniAppFileChooserDefault(var requestCode: Int) : MiniAppFileChooser {
...
+                val extensionWithoutDot =
+                    mime.trim { it <= ' ' }.substring(1, mime.trim { it <= ' ' }.length)
```
Comment (corycaywood) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFileChooser.kt:85]: You could simplify this by using `removePrefix(".")`?

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFileChooser.kt [hunk 5]
```diff
@@ -58,6 +70,32 @@ class MiniAppFileChooserDefault(var requestCode: Int) : MiniAppFileChooser {
...
+                val derivedMime = mtm.getMimeTypeFromExtension(extensionWithoutDot)
+                if (derivedMime != null && !results.contains(derivedMime)) {
+                    // adds valid mime type derived from the file extension
+                    results.add(derivedMime)
+                }
+            } else if (mtm.getExtensionFromMimeType(mime) != null && !results.contains(mime)) {
+                // adds valid mime type checked again file extensions mappings
+                results.add(mime)
```
Comment (corycaywood) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFileChooser.kt:93]: Shouldn't `mime` here be the trimmed one?

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFileChooser.kt [hunk 6]
```diff
@@ -58,6 +70,32 @@ class MiniAppFileChooserDefault(var requestCode: Int) : MiniAppFileChooser {
         return true
     }
 
+    /**
+     * Validation Utility for MimeTypes.
+     * @param mimeTypes The Mimetypes needs to convert to valid types.
+     */
+    @VisibleForTesting
+    internal fun extractValidMimeTypes(mimeTypes: Array<String>): List<String> {
+        val results: MutableList<String> = ArrayList()
```
Comment (corycaywood) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFileChooser.kt:79]: I think you don't need to create a temporary list here. You could instead use `mapNotNull`, for example:

```
mimeTypes.mapNotNull { mime -> 
  if ( // something //) {
    mime
  } else {
    null
  }
}
```

**Review by corycaywood — COMMENTED** [review]
It's looking good, but I left a few comments below.

Also, I think you're missing some unit test cases. For example the cases where a bad mime type is provided or duplicate types are provided.

**Review by corycaywood — APPROVED** [review]
Looks good! 👍

**Pull request merged**